lib/puppet/functions/dnsquery/lookup.rb

```diff
@@ -5,16 +5,22 @@
 # See the more specific functions if you only want one type returned.
 Puppet::Functions.create_function(:'dnsquery::lookup') do
   # @param domain the dns domain to lookup
+  # @param force_ipv6 ensure we get AAAA answers even if the requestor has no global ipv6 address
   # @param block an optional lambda to return a default value in case the lookup fails
   # @return An array of A and AAAA answers matching domain
   dispatch :dns_lookup do
     param 'Stdlib::Fqdn', :domain
+    optional_param 'Boolean', :force_ipv6
```

Comment:
I think it would be better for REFERENCE.md if it's not optional, but defaults to true or false. IMO it also should default to true since every other resolver works that way.

Reply:
thanks for the review

I agree i should default to true but it could be a bit to much of a change in behaviour for current users.  lets see what others say?

  As to optional, AFAIK you have to mark parameters as optional if they have a default, otherwise you could get linting errors (if nothing else)